From 3f36a0131953dcbcdf80d083e40b4f1cc2bc2165 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Tue, 2 May 2023 17:33:22 +0200
Subject: [PATCH 031/120] board: stm32mp2: add ft_board_setup

Activate OF_BOARD_SETUP and add the function ft_board_setup
to allow device tree modifcation by stm32mp2 board.

It is a preliminary step for MTD fixup support on
STMicroelectronics boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 788d87e2ee728c00eb218ba494553c4c1e8b8db4 board: stm32mp2: add ft_board_setup]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Iac0adcd2376c4372fef3b81c245e3edb9629dd8a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/268137
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 arch/arm/mach-stm32mp/Kconfig | 1 +
 board/st/stm32mp2/stm32mp2.c  | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm/mach-stm32mp/Kconfig b/arch/arm/mach-stm32mp/Kconfig
index 09b7d5123ae..fbcf502111f 100644
--- a/arch/arm/mach-stm32mp/Kconfig
+++ b/arch/arm/mach-stm32mp/Kconfig
@@ -86,6 +86,7 @@ config STM32MP25X
 	select ARM64
 	select CLK_STM32MP25
 	select OF_BOARD
+	select OF_BOARD_SETUP
 	select PINCTRL_STM32
 	select STM32_RCC
 	select STM32_RESET
diff --git a/board/st/stm32mp2/stm32mp2.c b/board/st/stm32mp2/stm32mp2.c
index 5db741ce24c..3a42a84e306 100644
--- a/board/st/stm32mp2/stm32mp2.c
+++ b/board/st/stm32mp2/stm32mp2.c
@@ -235,6 +235,11 @@ int board_late_init(void)
 	return 0;
 }
 
+int ft_board_setup(void *blob, struct bd_info *bd)
+{
+	return 0;
+}
+
 #if defined(CONFIG_USB_DWC3) && defined(CONFIG_CMD_STM32PROG_USB)
 #include <dfu.h>
 /*
-- 
2.47.2

